user/using-debug-vm.md

```diff
@@ -0,0 +1,169 @@
+you can restart builds in the debug mode to gain access to the actual VM or container
+(henceforth "debug VM").
+
+# Restarting a build to debug
+
+> This feature is available for private repositories and those public repositories for which
+> the feature is enabled.
+> To have the feature enabled on a public repository, please email us at
+> [support@travis-ci.com](mailto:support@travis-ci.com) indicating which ones.
+> The push access to the repository is also required.
+
+For private repositories, the "Debug build" or "Debug job" button is available on the upper right corner of
+the build/job page.
+
+![Screenshot of debug build/job buttons](/images/debug_buttons.png)
+
+For public repositories, an API call is required, in addition to the feature being enabled.
+
+## Starting a debug VM via API
+
+In addition to using the debug button, a debug build may be started via
+API (which is the only way to do it for public repositories).
+
+Here is an example for a public repository.
+Token should be set to your Travis CI token, and the job ID (`$id`) should point to the job you
+want to debug.
+
+```sh-session
+$ curl -s -X POST \
+  -H "Content-Type: application/json" \
+  -H "Accept: application/json" \
+  -H "Travis-API-Version: 3" \
+  -H "Authorization: token ********************" \
+  -d "{\"quiet\": true}" \
+  https://api.travis-ci.org/job/${id}/debug
+```
+
+For private repositories, the API endpoint is `https://api.travis-ci.com/job/${id}/debug`.
+
+The JSON payload could be empty (`-d "{}"`), in which case any command input and output
+you see while connected to the debug VM will be reflected on the build log.
+
+# Connecting to the debug VM
+
+Once the debug VM starts, it will go through the intial steps, including setting up language
+run times and caches (if applicable) and installing debug utilities.
+
+When the debug VM is ready for your connection, you will see:
+
+```
+Debug build initiated by BanzaiMan
+Setting up debug tools.
+Preparing debug sessions.
+Use the following SSH command to access the interactive debugging environment:
+ssh DwBhYvwgoBQ2dr7iQ5ZH34wGt@ny2.tmate.io
+This build is running in quiet mode. No session output will be displayed.
+This debug build will stay alive for 30 minutes.
+```
+
+Running the `ssh` command above will drop you in on a live VM.
+
+## Security considerations
+
+At this time, there is no authentication beyond the `ssh` command displayed in
+the logs.
+
+You may choose to share the `ssh` command with another person who might be
+able to help you out in debugging the build.
+
+However, this also means that anyone who can see the command while the debug
+VM is live can connect to the debug VM and potentially look at your secrets.
+This is the reason that the feature is not available by default on public
+repositories.
+
+Do note that when another `ssh` connection commences, that new connection
+will be attached to the session you are already in.
```

Comment:
Maybe we can add a little bit about what it means to have a shared session. That is to say, mention that the user can see what the other person is doing on the session. For example, if another person manages to dump the environment variables with `env`, the user will see that, too. (Thus giving an indication that the secrets are now at risk, and appropriate measures can be taken to minimize the damage.)